hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/TraceStatsEnricher.java

```diff
@@ -0,0 +1,46 @@
+package org.hypertrace.traceenricher.enrichment.enrichers;
+
+import static org.hypertrace.traceenricher.enrichedspan.constants.EnrichedSpanConstants.HEAD_EVENT_ID;
+import static org.hypertrace.traceenricher.enrichedspan.constants.EnrichedSpanConstants.TOTAL_NUMBER_OF_UNIQUE_API_NODES;
+
+import java.util.List;
+import org.hypertrace.core.datamodel.AttributeValue;
+import org.hypertrace.core.datamodel.Event;
+import org.hypertrace.core.datamodel.StructuredTrace;
+import org.hypertrace.core.datamodel.shared.ApiNode;
+import org.hypertrace.core.datamodel.shared.trace.AttributeValueCreator;
+import org.hypertrace.traceenricher.enrichment.AbstractTraceEnricher;
+import org.hypertrace.traceenricher.trace.util.ApiTraceGraph;
+import org.hypertrace.traceenricher.trace.util.ApiTraceGraphBuilder;
+
+public class TraceStatsEnricher extends AbstractTraceEnricher {
+
+  @Override
+  public void enrichTrace(StructuredTrace trace) {
+    ApiTraceGraph apiTraceGraph = ApiTraceGraphBuilder.buildGraph(trace);
+    if (apiTraceGraph.getApiNodeList().isEmpty()) {
+      return;
+    }
+    Event firstNodeHeadSpan = apiTraceGraph.getApiNodeList().get(0).getHeadEvent();
+    if (firstNodeHeadSpan == null) {
+      return;
+    }
+    addHeadSpanIdTraceAttribute(trace, firstNodeHeadSpan);
+    addTotalNumberOfUniqueApiNodesHeadSpanAttribute(
+        apiTraceGraph.getApiNodeList(), firstNodeHeadSpan);
+  }
+
+  private void addHeadSpanIdTraceAttribute(StructuredTrace trace, Event headSpan) {
+    AttributeValue attribute =
+        AttributeValue.newBuilder().setBinaryValue(headSpan.getEventId()).build();
+    trace.getAttributes().getAttributeMap().put(HEAD_EVENT_ID, attribute);
+  }
+
+  private void addTotalNumberOfUniqueApiNodesHeadSpanAttribute(
+      List<ApiNode<Event>> apiNodeList, Event headSpan) {
+    headSpan
+        .getEnrichedAttributes()
+        .getAttributeMap()
+        .put(TOTAL_NUMBER_OF_UNIQUE_API_NODES, AttributeValueCreator.create(apiNodeList.size()));
```

Comment:
Just a note (or Q): Are we looking at here the total number of API calls or the total number of unique APIs involved in the entire trace?
I think we should rename this to `TOTAL_NUMBER_OF_UNIQUE_API_NODES` -> `TOTAL_NUMBER_OF_API_NODES`. as these are not necessarily unique APIs. 
As an example, an API `(service: X API: GET /api/v1/u1)` can call same API of downstream service twice `(service Y : API : GET /api/v1/route?route_id=1234)` and `(service Y : API : GET /api/v1/route?route_id=2345)`. 
In this case, we will have three API nodes `(1 from service: X API: GET /api/v1/u1)` and two for Service Y's call.

If we are looking unique number of APIs, we will have to iterate over all the API nodes and check using api_id (enriched entity_id) to count.

Reply:
@kotharironak I believe, we need unique APIs nodes as (exit) calls count is captured in `num.api.exit.calls` attribute.

Reply:
@kotharironak will the code below work to get the `API_ID`? Was not sure which event to get it from. 

```
Set<String> uniqueApiIds = apiNodeList.stream()
        .map(eventApiNode -> EnrichedSpanUtils.getApiId(eventApiNode.getHeadEvent()))
        .collect(Collectors.toSet());
```
Is `API_ID` guaranteed to be there?

Reply:
If it runs after `EndpointEnricher`, it will have `API_ID` for all the events of ApiNode. See this - https://github.com/hypertrace/hypertrace-ingester/blob/main/hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/endpoint/EndpointEnricher.java#L161

Every API node with an entry API will have all its events enriched with API_ID. So, we can go with.
```
Set<String> uniqueApiIds = apiNodeList.stream()
        .filter(eventApiNode -> eventApiNode.getEntryApiBoundaryEvent().isPresent()
        .map(eventApiNode -> EnrichedSpanUtils.getApiId(eventApiNode.getEntryApiBoundaryEvent().get()))
        .collect(Collectors.toSet());
```

Reply:
Thanks @kotharironak. Makes sense